3rdparty/musl/CMakeLists.txt

```diff
@@ -31,80 +31,67 @@ set(MUSL_DIR ${CMAKE_CURRENT_BINARY_DIR}/musl)
+          ${MUSL_DIR}/include/execinfo.h
   CONFIGURE_COMMAND
-    ${CMAKE_COMMAND} -E chdir ${MUSL_DIR}
-    ${OE_BASH} -x ./configure
-      --includedir=${MUSL_INCLUDES}
-      CFLAGS=${MUSL_CFLAGS}
-      CC=${MUSL_CC}
-      CXX=${MUSL_CXX}
+    ${CMAKE_COMMAND} -E chdir ${MUSL_DIR} ${OE_BASH} -x ./configure
+    --includedir=${MUSL_INCLUDES} CFLAGS=${MUSL_CFLAGS} CC=${MUSL_CC}
+    CXX=${MUSL_CXX}
   BUILD_COMMAND
-    COMMAND ${CMAKE_COMMAND} -E copy_directory
-        ${MUSL_DIR}/include
-        ${MUSL_INCLUDES}
-    COMMAND ${CMAKE_COMMAND} -E copy_directory
-        ${MUSL_DIR}/arch/generic/bits
-        ${MUSL_INCLUDES}/bits
-    COMMAND ${CMAKE_COMMAND} -E copy_directory
-        ${MUSL_DIR}/arch/${ARCH}/bits
-        ${MUSL_INCLUDES}/bits
-    # bash -c requires the command string to be in a single line
-    COMMAND ${OE_BASH} -c "sed -f ${MUSL_DIR}/tools/mkalltypes.sed ${MUSL_DIR}/arch/${ARCH}/bits/alltypes.h.in ${MUSL_DIR}/include/alltypes.h.in > ${MUSL_INCLUDES}/bits/alltypes.h"
-    COMMAND ${CMAKE_COMMAND} -E copy
-        ${MUSL_DIR}/arch/${ARCH}/bits/syscall.h.in
-        ${MUSL_INCLUDES}/bits/syscall.h
-    # bash -c requires the command string to be in a single line
-    COMMAND ${OE_BASH} -c "sed -n -e s/__NR_/SYS_/p < ${MUSL_DIR}/arch/${ARCH}/bits/syscall.h.in >> ${MUSL_INCLUDES}/bits/syscall.h"
-    COMMAND ${CMAKE_COMMAND} -E copy
-      ${MUSL_INCLUDES}/endian.h
-      ${MUSL_INCLUDES}/__endian.h
-    COMMAND ${CMAKE_COMMAND} -E copy
-      ${PATCHES_DIR}/endian.h
-      ${MUSL_INCLUDES}/endian.h
-    # Append deprecations.h to all C header files.
-    COMMAND ${OE_BASH} -c "${MUSL_APPEND_DEPRECATIONS}"
-    # Copy local deprecations.h to include/bits/deprecated.h.
-    COMMAND ${CMAKE_COMMAND} -E copy
-      ${CMAKE_CURRENT_LIST_DIR}/deprecations.h
-      ${MUSL_INCLUDES}/bits/deprecations.h
-  BUILD_BYPRODUCTS
-    ${MUSL_INCLUDES} ${MUSL_DIR}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${MUSL_DIR}/include
+          ${MUSL_INCLUDES}
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${MUSL_DIR}/arch/generic/bits
+          ${MUSL_INCLUDES}/bits
+  COMMAND ${CMAKE_COMMAND} -E copy_directory ${MUSL_DIR}/arch/${ARCH}/bits
+          ${MUSL_INCLUDES}/bits
+  # bash -c requires the command string to be in a single line
+  COMMAND
+    ${OE_BASH} -c
+    "sed -f ${MUSL_DIR}/tools/mkalltypes.sed ${MUSL_DIR}/arch/${ARCH}/bits/alltypes.h.in ${MUSL_DIR}/include/alltypes.h.in > ${MUSL_INCLUDES}/bits/alltypes.h"
+  COMMAND ${CMAKE_COMMAND} -E copy ${MUSL_DIR}/arch/${ARCH}/bits/syscall.h.in
+          ${MUSL_INCLUDES}/bits/syscall.h
+  # bash -c requires the command string to be in a single line
+  COMMAND
+    ${OE_BASH} -c
+    "sed -n -e s/__NR_/SYS_/p < ${MUSL_DIR}/arch/${ARCH}/bits/syscall.h.in >> ${MUSL_INCLUDES}/bits/syscall.h"
+  COMMAND ${CMAKE_COMMAND} -E copy ${MUSL_INCLUDES}/endian.h
+          ${MUSL_INCLUDES}/__endian.h
+  COMMAND ${CMAKE_COMMAND} -E copy ${PATCHES_DIR}/endian.h
+          ${MUSL_INCLUDES}/endian.h
```

Comment:
I don't understand why copies like this are needed at all.   Can't we just put ${PATCHES_DIR} in the include path before ${MUSL_INCLUDES} and avoid all this busy work?

Reply:
After the patches have been applied by copying, we run MUSL's _configure_ command which depends on files being in certain locations. Therefore, we have to apply the patches as we do now.
Post _configure_, we further copy files to the relevant _output_ folders for use by other modules as well as for packaging for installation. These too require files to be in specific places.

Note: MUSL itself does not come with cmake support and relies on linux specific commands to build. We have implemented this bit of CMake scripting to integrate it into OE's cmake based build system. This has the additional benefit that MUSL based enclaves can also be built on Windows.